File: turb/RANS_Channel-master/matlab/velTransSLS.m
%**************************************************************************
%       function, transform of van Driest velocity to van Driest extended 
%       velocity
%**************************************************************************
%   dustar = [1+(y/Rets)dRets/dy] duvd
%
% Inputs:
%   uvd     velocity with van Driest transformation
%   r       density
%   mu      molecular viscosity
%   mesh    mesh structure
%
% Output:
%   ustar   velocity with the extended van Driest transformation
%
function [ustar] = velTransSLS(uvd,ReTst,mesh)

    n = size(uvd,1);

    % Calculating correction factor of semi-local velocoty transformation
    dRTSdy = mesh.ddy*ReTst;
    fact = 1 + mesh.y./ReTst.*dRTSdy;

    ustar = zeros(n,1);
    for i=2:n
        ustar(i) = ustar(i-1) + 0.5*(fact(i)+fact(i-1))*(uvd(i)-uvd(i-1));
    end
    
end

File: turb/RANS_Channel-master/python/velTransSLS.py
# -*- coding: utf-8 -*-
"""
Created on Fri Jan  4 14:03:22 2019

@author: gjoterorodrigu
"""

def velTransSLS(u,r,mu,ReTau,mesh):

    import numpy as np
    from velTransVD import velTransVD

    n = mesh.nPoints

    # Calculating correction factor of semi-local velocoty transformation
    ReTauStar = ReTau*np.sqrt(r/r[0])/(mu/mu[0])
    dRTSdy    = mesh.ddy@ReTauStar
    fact      = 1 + mesh.y/ReTauStar*dRTSdy

    ystar = mesh.y*ReTauStar
    ypl, uvd = velTransVD(u,r,ReTau,mesh)
    
    ustar = np.zeros(n)
    for i in range(1,n):
        ustar[i] = ustar[i-1] + 0.5*(fact[i]+fact[i-1])*(uvd[i]-uvd[i-1])

    return ystar,ustar
